src/imagery/i.eodag/i.eodag.py

```diff
@@ -0,0 +1,319 @@
+#!/usr/bin/env python3
+
+############################################################################
+#
+# MODULE:      i.eodag
+#
+# AUTHOR(S):   Hamed Elgizery
+# MENTOR(S):   Luca Delucchi, Veronica Andreo, Stefan Blumentrath
+#
+# PURPOSE:     Downloads imagery datasets e.g. Landsat, Sentinel, and MODIS
+#              using EODAG API.
+# COPYRIGHT:   (C) 2024-2025 by Hamed Elgizery, and the GRASS development team
+#
+#              This program is free software under the GNU General Public
+#              License (>=v2). Read the file COPYING that comes with GRASS
+#              for details.
+#
+#############################################################################
+
+
+# %Module
+# % description: Eodag interface to install imagery datasets from various providers.
+# % keyword: eodag
+# % keyword: sentinel
+# % keyword: landsat
+# % keyword: modis
+# % keyword: datasets
+# % keyword: download
+# %end
+
+# %option
+# % key: dataset
+# % type: string
+# % description: Imagery dataset to search for
+# % required: yes
+# % answer: S1_SAR_GRD
+# % guisection: Filter
+# %end
+
+# %option G_OPT_M_DIR
+# % key: output
+# % description: Name for output directory where to store downloaded data OR search results
+# % required: no
+# % guisection: Output
+# %end
+
+# %option
+# % key: id
+# % type: string
+# % multiple: yes
+# % description: List of scenes IDs to download
+# % guisection: Filter
+# %end
+
+# %option
+# % key: provider
+# % type: string
+# % description: Provider to use for searching/downloading, if none is provided the searching will be done according to the config file priority
+# % required: no
+# % guisection: Filter
+# %end
+
+# %option
+# % key: start
+# % type: string
+# % description: Start date ('YYYY-MM-DD'), by default it is 60 days ago
+# % guisection: Filter
+# %end
+
+# %option
+# % key: end
+# % type: string
+# % description: End date ('YYYY-MM-DD')
+# % guisection: Filter
+# %end
+
+# %option
+# % key: relation
+# % type: string
+# % description: Relation with area of interest
+# % options: intersects, contains, within
+# % answer: intersects
+# % guisection: Optional
+# %end
+
+# %option
+# % key: clouds
+# % type: integer
+# % description: Maximum cloud cover percentage for Landsat scene
+# % required: no
+# % guisection: Filter
+# %end
+
+# %option
+# % key: timeout
+# % type: integer
+# % description: Download timeout in seconds
+# % answer: 300
+# % guisection: Optional
+# %end
+
+
+# %flag
+# % key: l
+# % description: List the search result without downloading
```

Comment:
There will be probably several types of information to print. It may make sense to have here options instead of a flag....

Reply:
Yes, @ninsbl do you have an output of i.sentinel.download to show to @HamedElgizery ?

Reply:
Further below an output from i.sentinel.downoad.

What I think could be different information that should be possible to print to user are:
- current eodag configuration (an file location)
- configuration of a chosen provider (e.g. when print option and provider option are given)
- available providers
- available providers for a chosen product (if only a product is chosen)
- available products
- available products of a chosen provider (e.g. when print option and provider option are given)
- queryables / metadata of product types (requires both provider and product option).

There may be more. But printable data has different complexity (can be a simple list where plain output would be sufficient or a hirarchical structure (e.g. product metadata) where JSON output would be more appropriate. But I would say printing is a next step when basic query and download functionality is more mature...
```
"29bbec5a-9ab4-41e5-88f5-d716242d944f S3B_SL_1_RBT____20240526T195554_20240526T195854_20240528T010235_0179_093_242_1080_PS2_O_NT_004 2024-05-26T19:55:54Z cloudcover_NA SL_1_RBT___ 417.02 MB\nf702eab4-1644-447b-8046-34e09b5dd250 S3B_SL_1_RBT____20240526T181455_20240526T181755_20240528T010629_0179_093_241_1080_PS2_O_NT_004 2024-05-26T18:14:54Z cloudcover_NA SL_1_RBT___ 425.36 MB\naed9bd3f-e134-45ef-8b33-576523312296 S3B_SL_1_RBT____20240526T195254_20240526T195554_20240528T010205_0179_093_242_0900_PS2_O_NT_004 2024-05-26T19:52:54Z cloudcover_NA SL_1_RBT___ 404.04 MB\nad03b03b-44e8-41d8-abb9-6713c833d325 S3B_SL_1_RBT____20240526T213353_20240526T213653_20240528T010044_0179_093_243_0900_PS2_O_NT_004 2024-05-26T21:33:53Z cloudcover_NA SL_1_RBT___ 391.18 MB\n8d55c4fc-662c-411b-87ba-3b3029826e0d S3A_SL_1_RBT____20240526T171249_20240526T171549_20240528T031123_0179_112_383_1080_PS1_O_NT_004 2024-05-26T17:12:48Z cloudcover_NA SL_1_RBT___ 424.19 MB\n227018c9-87f7-4bef-b564-40257ab7a4c2 S3A_SL_1_RBT____20240526T171549_20240526T171849_20240528T031123_0179_112_383_1260_PS1_O_NT_004 2024-05-26T17:15:48Z cloudcover_NA SL_1_RBT___ 425.10 MB\n96674eba-7e6c-4020-ace0-f4c8ee355467 S3A_SL_1_RBT____20240526T185348_20240526T185648_20240528T044547_0179_112_384_1080_PS1_O_NT_004 2024-05-26T18:53:47Z cloudcover_NA SL_1_RBT___ 425.44 MB\n7b52e1cb-4bca-46fb-8642-224166705c46 S3A_SL_1_RBT____20240526T203147_20240526T203447_20240528T062503_0179_112_385_0900_PS1_O_NT_004 2024-05-26T20:31:47Z cloudcover_NA SL_1_RBT___ 405.00 MB\n6392a06d-ba7a-4a7d-aad3-b2732edcc21f S3A_SL_1_RBT____20240526T203447_20240526T203747_20240528T062504_0179_112_385_1080_PS1_O_NT_004 2024-05-26T20:34:47Z cloudcover_NA SL_1_RBT___ 406.67 MB\nfa9475ae-aebb-4760-a674-bba5335da40a S3B_SL_1_RBT____20240527T093548_20240527T093848_20240528T102505_0179_093_250_1800_PS2_O_NT_004 2024-05-27T09:35:48Z cloudcover_NA SL_1_RBT___ 448.52 MB\n0a900a66-9b06-4a16-aadf-c85eba792814 S3B_SL_1_RBT____20240527T111347_20240527T111647_20240528T102940_0180_093_251_1620_PS2_O_NT_004 2024-05-27T11:13:47Z cloudcover_NA SL_1_RBT___ 425.28 MB\n86361c46-41af-4553-bf31-3ceaf541634f S3B_SL_1_RBT____20240527T111947_20240527T112247_20240528T103038_0179_093_251_1980_PS2_O_NT_004 2024-05-27T11:19:47Z cloudcover_NA SL_1_RBT___ 432.68 MB\nffd76101-4ff0-45bb-bfe4-b47175b1a61e S3B_SL_1_RBT____20240527T111647_20240527T111947_20240528T103009_0179_093_251_1800_PS2_O_NT_004 2024-05-27T11:16:47Z cloudcover_NA SL_1_RBT___ 422.23 MB\nba940202-dd53-4e7c-908e-a7cf0b5b79d2 S3B_SL_1_RBT____20240527T192644_20240527T192944_20240528T103921_0180_093_256_0900_PS2_O_NT_004 2024-05-27T19:26:43Z cloudcover_NA SL_1_RBT___ 400.10 MB\n99240ed7-0449-4b91-bca1-07160b7e50a0 S3B_SL_1_RBT____20240527T192944_20240527T193244_20240528T103942_0179_093_256_1080_PS2_O_NT_004 2024-05-27T19:29:43Z cloudcover_NA SL_1_RBT___ 424.03 MB\nb7b763c6-6a3b-4376-87a1-478c22031720 S3B_SL_1_RBT____20240527T175144_20240527T175444_20240528T104434_0179_093_255_1260_PS2_O_NT_004 2024-05-27T17:51:44Z cloudcover_NA SL_1_RBT___ 417.55 MB\nd72f5d69-ddc7-447f-893c-3bb95531a8ef S3B_SL_1_RBT____20240527T093848_20240527T094148_20240528T102533_0179_093_250_1980_PS2_O_NT_004 2024-05-27T09:38:48Z cloudcover_NA SL_1_RBT___ 442.27 MB\nf50eed12-42ad-4c01-8790-20c6aa8fc8e8 S3B_SL_1_RBT____20240527T174844_20240527T175144_20240528T104415_0179_093_255_1080_PS2_O_NT_004 2024-05-27T17:48:44Z cloudcover_NA SL_1_RBT___ 425.20 MB\n174ae551-1b7e-4168-aac8-5af4df6dd5ff S3B_SL_1_RBT____20240527T211043_20240527T211343_20240528T144646_0179_093_257_1080_PS2_O_NT_004 2024-05-27T21:10:42Z cloudcover_NA SL_1_RBT___ 402.48 MB\n874b59da-443d-4ee7-b814-e9c84a3e43c7 S3B_SL_1_RBT____20240527T210743_20240527T211043_20240528T144626_0179_093_257_0900_PS2_O_NT_004 2024-05-27T21:07:42Z cloudcover_NA SL_1_RBT___ 395.54 MB\n300facba-66dc-4fd3-a42a-e4f8d9e29dbb S3A_SL_1_RBT____20240527T083342_20240527T083642_20240528T181029_0180_113_007_1800_PS1_O_NT_004 2024-05-27T08:33:41Z cloudcover_NA SL_1_RBT___ 442.87 MB\n41385ccf-9068-4041-ba2b-64fa6b158feb S3A_SL_1_RBT____20240527T101141_20240527T101441_20240528T195034_0179_113_008_1620_PS1_O_NT_004 2024-05-27T10:11:41Z cloudcover_NA SL_1_RBT___ 429.01 MB\nf44cf156-78d4-45c2-9cde-b5e0301662bc S3A_SL_1_RBT____20240527T101741_20240527T102041_20240528T195035_0179_113_008_1980_PS1_O_NT_004 2024-05-27T10:17:41Z cloudcover_NA SL_1_RBT___ 453.01 MB\n18831c4d-f223-4abf-8248-703056117101 S3A_SL_1_RBT____20240527T101441_20240527T101741_20240528T195035_0179_113_008_1800_PS1_O_NT_004 2024-05-27T10:14:41Z cloudcover_NA SL_1_RBT___ 450.67 MB\n"
```

Reply:
Handled further here: [here](https://github.com/HamedElgizery/grass-addons/issues/4)